### PR TITLE
No longer assume that the `UnicastBusConfig` is present 

### LINKED ISSUE #2913: `DelayedDeliveryFeature` makes assumptions about `UnicastBusConfig` begin present
## Who's affected
Developers trying out develop 6.0.0-unstable0878 and later without a `UnicastBusConfig` defined (i.ex. self-hosting)

## Symptons
1. Starting an endpoint without a `UnicastBusConfig` throw a `NullReferenceException`

## Steps to reproduce
1. Take [Startup Shutdown Sequence Sample](http://docs.particular.net/samples/startup-shutdown-sequence/) and upgrade it to latest v6 develop
2. Start the endpoint

```
System.NullReferenceException was unhandled
  HResult=-2147467261
  Message=Object reference not set to an instance of an object.
  Source=NServiceBus.Core
  StackTrace:
       at NServiceBus.Features.DelayedDeliveryFeature.<>c.<.ctor>b__0_0(SettingsHolder s) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\DelayedDelivery\DelayedDeliveryFeature.cs:line 19
       at NServiceBus.Features.FeatureActivator.SetupFeatures(FeatureConfigurationContext context) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Features\FeatureActivator.cs:line 103
       at NServiceBus.Configure.Initialize() in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Configure.cs:line 97
       at NServiceBus.Bus.Create(BusConfiguration configuration) in C:\BuildAgent\work\3206e2123f54fce4\src\NServiceBus.Core\Bus.cs:line 17
       at Program.Main() in C:\particular\docs.particular.net\samples\startup-shutdown-sequence\Version_6\Sample\Program.cs:line 20
  InnerException: 
```

### PR DESCRIPTION
**Pull request by danielmarbach**
fixes #2913

### CODE REVIEW

**Comment by danielmarbach** [comment]
@Particular/nservicebus please review

File: src/NServiceBus.Core/Unicast/Config/UnicastBus.cs [hunk 1]
```diff
@@ -20,7 +20,7 @@ internal UnicastBus()
             Defaults(s =>
             {
                 var section = s.GetConfigSection<UnicastBusConfig>();
-                if (section.TimeoutManagerAddress != null)
+                if (section?.TimeoutManagerAddress != null)
```
Comment (SzymonPobiega) [inline src/NServiceBus.Core/Unicast/Config/UnicastBus.cs]: Nitpicking: how about
```
var address = section?.TimeoutManagerAddress
if (address != null) {
   s.Set("TimeoutManagerAddress", address);
}
```
one access to section less (not for performance reasons, more about styling)

File: src/NServiceBus.Core/DelayedDelivery/DelayedDeliveryFeature.cs [hunk 2]
```diff
@@ -16,7 +16,7 @@ public DelayedDeliveryFeature()
             DependsOnOptionally<TimeoutManager>();
             Defaults(s =>
             {
-                var timeoutManagerAddressConfiguration = new TimeoutManagerAddressConfiguration(s.GetConfigSection<UnicastBusConfig>().TimeoutManagerAddress);
+                var timeoutManagerAddressConfiguration = new TimeoutManagerAddressConfiguration(s.GetConfigSection<UnicastBusConfig>()?.TimeoutManagerAddress);
```
Comment (SzymonPobiega) [inline src/NServiceBus.Core/DelayedDelivery/DelayedDeliveryFeature.cs]: Nice :+1:

Comment (danielmarbach) [inline src/NServiceBus.Core/DelayedDelivery/DelayedDeliveryFeature.cs]: @szymonpobiega I'll do.

**Comment by danielmarbach** [comment]
@SzymonPobiega applied and rebased

**Pull request merged**